frontend/src/app/components/MobileDialog/index.js

```diff
@@ -1,284 +1,308 @@
-      if (response.ok) {
-        sendToGA("event", {
-          eventCategory: "ExperimentDetailsPage Interactions",
-          eventAction: "button click",
-          eventLabel: "link sent to email"
-        });
-      }
-      this.setState({
-        isSuccess: response.ok,
-        isError: !response.ok
-      });
-    });
-  }
-
-  reset = (e: Object) => {
-    e.preventDefault();
-    e.stopPropagation();
-    this.setState(DEFAULT_STATE);
-    this.fetchCountryCode();
-  }
-
-  close = () => {
-    if (this.props.onCancel) {
-      this.props.onCancel();
-      this.props.sendToGA("event", {
-        eventCategory: "ExperimentDetailsPage Interactions",
-        eventAction: "button click",
-        eventLabel: "cancel Send link to device dialog"
-      });
-    }
-  }
-
-  handleKeyDown(e: Object) {
-    if (e.key === "Escape") this.close();
-  }
-}
+// @flow
+import React from "react";
+import classnames from "classnames";
+// $FlowFixMe
+import { isValidNumber } from "libphonenumber-js";
+// $FlowFixMe
+import { validate } from "email-validator";
+import { Localized } from "fluent-react/compat";
+import LocalizedHtml from "../LocalizedHtml";
+import Loading from "../Loading";
+
+import "./index.scss";
+
+import { subscribeToBasket, subscribeToBasketSMS, acceptedSMSCountries } from "../../lib/utils";
+
+type MobileDialogProps = {
+  getWindowLocation: Function,
+  onCancel: Function,
+  sendToGA: Function,
+  experiment: Object,
+  fetchCountryCode: Function,
+  fromFeatured?: boolean
+}
+
+type MobileDialogState = {
+  loading: boolean,
+  isSuccess: boolean,
+  isError: boolean,
+  allowSMS: boolean,
+  submitAttempted: boolean,
+  recipient: string,
+  country: string
+}
+
+const DEFAULT_STATE = {
+  loading: true,
+  isSuccess: false,
+  isError: false,
+  allowSMS: false,
+  submitAttempted: false,
+  recipient: "",
+  country: "US"
+};
+
+export default class MobileDialog extends React.Component {
+  props: MobileDialogProps
+  state: MobileDialogState
+
+  modalContainer: Object
+
+  constructor(props: MobileDialogProps) {
+    super(props);
+    this.state = DEFAULT_STATE;
+  }
+
+  componentDidMount() {
+    this.focusModalContainer();
+    this.fetchCountryCode();
+  }
+
+  fetchCountryCode() {
+    this.props.fetchCountryCode()
+      .then((resp) => resp.json())
+      .then((data) => {
+        const country = data.country_code;
+        if (acceptedSMSCountries.includes(country)) {
+          this.setState({
+            loading: false,
+            allowSMS: true,
+            country: country
+          });
+        } else this.setState({ loading: false });
+      }).catch((e) => this.setState({ loading: false }));
+  }
+
+  render() {
+    const { experiment, sendToGA, fromFeatured } = this.props;
+    const { title, android_url, ios_url } = experiment;
+    const { isSuccess, allowSMS, loading } = this.state;
+
+    const handleAppLinkClick = () => {
+      const platform = ios_url ? "ios" : "android";
```

Comment:
I'm confused here (which is fine because I have zero conflict, i'm just randomly looking at PRs like a maniac), but [how] does this work if you have both an iOS _and_ an Android app available on TestPilot? If your manifest has both `android_url` and `ios_url` defined in the YAML, I don't think you'd ever get the `"android"` case here. Maybe it doesn't matter, just something my eyes cannot unsee.

Reply:
we have yet to handle that case, I believe there are plans to handle that in the future cc/ @johngruen